cpu/stm32_common/periph/eeprom.c

```diff
@@ -45,7 +101,7 @@ size_t eeprom_read(uint32_t pos, uint8_t *data, size_t len)
     DEBUG("Reading data from EEPROM at pos %" PRIu32 ": ", pos);
     for (size_t i = 0; i < len; i++) {
         _wait_for_pending_operations();
-        *p++ = *(uint8_t *)(EEPROM_START_ADDR + pos++);
+        *p++ = *(__IO uint8_t *)(EEPROM_START_ADDR + pos++);
```

Comment:
This should go in its own commit as it's a bit unrelated. It also requires test on other supported families...